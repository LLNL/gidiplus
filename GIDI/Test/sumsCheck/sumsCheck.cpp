/*
# <<BEGIN-copyright>>
# Copyright 2019, Lawrence Livermore National Security, LLC.
# See the top-level COPYRIGHT file for details.
# 
# SPDX-License-Identifier: MIT
# <<END-copyright>>
*/

#include <stdlib.h>
#include <iostream>

#include <GIDI_testUtilities.hpp>

static char const *description = "Prints the label for each **crossSectionSum** and **multiplicitySum** node.";

void main2( int argc, char **argv );
/*
=========================================================
*/
int main( int argc, char **argv ) {

    try {
        main2( argc, argv ); }
     catch (std::exception &exception) {
        std::cerr << exception.what( ) << std::endl;
        exit( EXIT_FAILURE ); }
    catch (char const *str) {
        std::cerr << str << std::endl;
        exit( EXIT_FAILURE ); }
    catch (std::string &str) {
        std::cerr << str << std::endl;
        exit( EXIT_FAILURE );
    }

    exit( EXIT_SUCCESS );
}
/*
=========================================================
*/
void main2( int argc, char **argv ) {

    argvOptions argv_options( "sumsCheck", description );
    ParseTestOptions parseTestOptions( argv_options, argc, argv );

    parseTestOptions.parse( );

    GIDI::Construction::PhotoMode photo_mode = parseTestOptions.photonMode( GIDI::Construction::PhotoMode::nuclearAndAtomic );
    GIDI::Construction::Settings construction( GIDI::Construction::ParseMode::all, photo_mode );
    PoPI::Database pops;
    GIDI::Protare *protare = parseTestOptions.protare( pops, "../pops.xml", "../all.map", construction, PoPI::IDs::neutron, "O16" );

    GIDI::ProtareSingle *protareSingle( protare->protare( 0 ) );

    std::cout << stripDirectoryBase( protareSingle->fileName( ), "/GIDI/Test/" ) << std::endl;

    GIDI::Sums::Sums const &sums( protareSingle->sums( ) );

    std::cout << std::endl;
    std::cout << "crossSectionSum labels:" << std::endl;
    GIDI::Suite const &crossSectionSums = sums.crossSectionSums( );
    for( std::size_t index = 0; index < crossSectionSums.size( ); ++index ) {
        GIDI::Sums::CrossSectionSum const *crossSectionSum = crossSectionSums.get<GIDI::Sums::CrossSectionSum>( index );

        std::cout << "    " << crossSectionSum->label( ) << std::endl;
    }

    std::cout << std::endl;
    std::cout << "multiplicitySum labels:" << std::endl; 
    GIDI::Suite const &multiplicitySums = sums.multiplicitySums( );
    for( std::size_t index = 0; index < multiplicitySums.size( ); ++index ) {
        GIDI::Sums::MultiplicitySum const *multiplicitySum = multiplicitySums.get<GIDI::Sums::MultiplicitySum>( index );

        std::cout << "    " << multiplicitySum->label( ) << std::endl;
    }
//    for( auto iter = sums.multiplicitySums( ).begin( ); iter != sums.multiplicitySums( ).end( ); iter++ ) std::cout << "    " << (*iter)->label( ) << std::endl;

    delete protare;
}
