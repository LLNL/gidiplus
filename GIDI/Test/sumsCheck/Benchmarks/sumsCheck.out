/.../GIDI/Test/Data/MG_MC.XML/neutrons/n-008_O_016.xml

crossSectionSum labels:
    total
    nonelastic
    (z,n)
    (z,p)
    (z,d)
    (z,t)
    (z,alpha)

multiplicitySum labels:
    2n + O15 + photon total gamma multiplicity
    O17 + photon total gamma multiplicity
    (z,n) total gamma multiplicity
    (z,p) total gamma multiplicity
    (z,d) total gamma multiplicity
    (z,t) total gamma multiplicity
    (z,alpha) total gamma multiplicity
